airbyte-cdk/bulk/toolkits/load-db/src/main/kotlin/io/airbyte/cdk/load/orchestration/db/direct_load_table/DirectLoadTableStreamLoader.kt

```diff
@@ -82,63 +120,74 @@ class DirectLoadTableAppendTruncateStreamLoader(
     private val streamStateStore: StreamStateStore<DirectLoadTableExecutionConfig>,
 ) : StreamLoader {
     // can't use lateinit because of weird kotlin reasons.
-    // this field is always overwritten in start().
-    private var writingToTempTable: Boolean = false
+    /**
+     * Indicates whether we're writing to the temporary table or directly to the real table. This is
+     * determined during start() based on table states and generation IDs.
+     * - true: Writing to temp table, will need to copy/overwrite to real table later
+     * - false: Writing directly to real table, no additional action needed at close
+     */
+    private var isWritingToTemporaryTable: Boolean = false
 
     override suspend fun start() {
+        logger.info { "AppendTruncateStreamLoader starting for stream: ${stream.descriptor}" }
+
         if (initialStatus.tempTable != null) {
-            if (
-                initialStatus.tempTable.isEmpty ||
-                    nativeTableOperations.getGenerationId(tempTableName) >=
-                        stream.minimumGenerationId
-            ) {
+            val generationId = nativeTableOperations.getGenerationId(tempTableName)
```

Comment:
I'm not sure this is safe. IIRC getGenerationId assumes that the table is nonempty? (we can change that contract, but then I think we can actually stop fetching the isEmpty value entirely, which is a bigger change)